sqlengine/postgres_engine.go

```diff
@@ -37,6 +40,18 @@ func (d *PostgresEngine) Open(address string, port int64, dbname string, usernam
 	return nil
 }
 
+func (d *PostgresEngine) Ping() error {
+	err := d.db.Ping()
+	if err != nil {
+		d.logger.Error("sql-ping-error", err)
+		if strings.Contains(err.Error(), "password authentication failed") {
```

Comment:
It's not good practice to use string matching to check errors. It would be better to use a [type assertion](https://golang.org/ref/spec#Type_assertions) to cast the error to a `*pq.Error`, and then check the error code directly.